File: pertemuan 10/DoublyLinkedListQueue.py
from Node import Node

class DoublyLinkedListQueue:
    def _init_(self):
        self.head = None
        self.tail = None

    def tambah_data(self, nomor_antrian, nama):
        new_node = Node(nomor_antrian, nama)
        if not self.head:
            self.head = new_node
            self.tail = new_node
        else:
            new_node.prev = self.tail
            self.tail.next = new_node
            self.tail = new_node

    def hapus_data(self):
        if not self.head:
            return None
        nama_selesai = self.head.nama
        self.head = self.head.next
        if self.head:
            self.head.prev = None
        return nama_selesai

    def daftar_penerima_vaksin(self):
        current = self.head
        if not current:
            print("Antrian kosong")
        else:
            print("Daftar Penerima Vaksin")
            print("| No.  | Nama   |")
            while current:
                print(f"| {current.nomor_antrian:<5} | {current.nama:<6} |")
                current = current.next

File: pertemuan 10/DoublyLinkedListQueue.java
public class DoublyLinkedListQueue {
    private Node head;
    private Node tail;

    public DoublyLinkedListQueue() {
        this.head = null;
        this.tail = null;
    }

    public void tambahData(int nomorAntrian, String nama) {
        Node newNode = new Node(nomorAntrian, nama);
        if (head == null) {
            head = newNode;
            tail = newNode;
        } else {
            newNode.prev = tail;
            tail.next = newNode;
            tail = newNode;
        }
    }

    public String hapusData() {
        if (head == null) {
            return null;
        }
        String namaSelesai = head.nama;
        head = head.next;
        if (head != null) {
            head.prev = null;
        }
        return namaSelesai;
    }

    public void daftarPenerimaVaksin() {
        Node current = head;
        if (current == null) {
            System.out.println("Antrian kosong");
        } else {
            System.out.println("Daftar Penerima Vaksin");
            System.out.println("| No.  | Nama   |");
            while (current != null) {
                System.out.printf("| %-5d | %-6s |\n", current.nomorAntrian, current.nama);
                current = current.next;
            }
        }
    }
}